File: frontend/src/types/edna.ts
export interface EDNASample {
  id: string;
  sampleCode: string;
  stationId: string;
  cruiseId: string;
  samplingDate: string;
  latitude: number;
  longitude: number;
  samplingDepthMeters: number;
  filterType: 'Sterivex 0.22µm' | 'Cellulose Nitrate 0.45µm' | 'Glass Fiber';
  waterVolumeFilteredLiters: number;
  dnaYieldNgPerUl: number;
  targetMarkers: ('12S' | '16S' | 'COI' | '18S')[];
  sequencingPlatform: 'Illumina NovaSeq' | 'Illumina MiSeq' | 'Oxford Nanopore';
  totalDetectionsCount: number;
  collectorName: string;
}

export interface EDNADetection {
  id: string;
  sampleId: string;
  sampleCode: string;
  speciesId?: string;
  scientificName: string;
  commonName: string;
  taxonomicRank: 'Species' | 'Genus' | 'Family';
  markerUsed: '12S' | '16S' | 'COI' | '18S';
  readCount: number;
  relativeAbundancePercent: number;
  matchIdentityPercent: number; // e.g. 99.4%
  confidenceScore: number; // 0.0 - 1.0 (e.g. 0.98)
  referenceDatabase: 'NCBI BLAST' | 'MIFish' | 'BOLD Systems';
  samplingCoordinates: [number, number];
}

File: backend/app/schemas/edna.py
"""
eDNA Schemas.
"""

from typing import Any, Dict, Optional
from pydantic import BaseModel


class EDNASampleResponse(BaseModel):
    id: str
    dataset_id: Optional[str] = None
    sample_code: Optional[str] = None
    latitude: float
    longitude: float
    depth: Optional[float] = None
    collected_at: Optional[str] = None
    sequencing_platform: Optional[str] = None
    metadata: Optional[Dict[str, Any]] = None


class EDNADetectionResponse(BaseModel):
    id: str
    sample_id: str
    species_id: Optional[str] = None
    scientific_name: Optional[str] = None
    read_count: Optional[int] = None
    confidence_score: Optional[float] = None
    metadata: Optional[Dict[str, Any]] = None
